File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/kafka/ConsumerUserStream.java
package edu.uci.ics.cloudberry.noah.kafka;

import edu.uci.ics.cloudberry.noah.feed.CmdLineAux;
import edu.uci.ics.cloudberry.noah.feed.Config;
import org.kohsuke.args4j.CmdLineException;

public class ConsumerUserStream {
    public static void main(String[] args) {

        try {
            Config config = CmdLineAux.parseCmdLine(args);

            if (config.getKafkaServer().isEmpty() || config.getKafkaId().isEmpty()) {
                throw new CmdLineException("Should provide a server (hostname:port) and a consumer ID");
            }

            ConsumerKafka consumer = new ConsumerKafka();
            String[] topics = {"TwitterUserStream"};
            consumer.run(config.getKafkaServer(), config.getKafkaId(), topics);

        } catch (CmdLineException e) {
            e.printStackTrace(System.err);
        }
    }
}

File: noah/src/main/scala/edu/uci/ics/cloudberry/noah/kafka/ConsumerUserStream.scala
package edu.uci.ics.cloudberry.noah.kafka

import edu.uci.ics.cloudberry.noah.feed.{CmdLineAux, Config}
import org.kohsuke.args4j.CmdLineException

/**
  * Created by Monique on 7/18/2016.
  */

object ConsumerUserStream extends GeneralTopicConsumer{
  def main(args: Array[String]) {
    runMain(args, Config.Source.User)
  }
}
